### PR TITLE
ner analyzer changes , tokenizer initialization updated .

### PR DESCRIPTION
**Pull request by akar5h**
Apart from the tokenizer fix , I also observed one more thing, in the documentation of configure Ner Analyzer it is mentioned that  _"# NER analyzer does not need configuration settings" ._ 
But while predicting the NER tags using NERAnalyzer's **analyze_input** method, analyzer_config is required (below). Hence the NER tagging fails. 

```
        if analyzer_config is None:
             raise ValueError("analyzer_config can't be None")
``` 

So two things can be done, either we mention config setting for analyzer 
or fix the batch size 1 in code,
I have done a quick fix here but I need your comment so I can make a cleaner fix. Either make the ner's **analyze_input** batch free or define a config and a batch size for ners .

### CODE REVIEW

**Comment by akar5h** [comment]
fix for #151

File: obsei/analyzer/ner_analyzer.py
Comment (lalitpagaria) [inline obsei/analyzer/ner_analyzer.py:38]: Two times `tokenizer` is assigned. One assignment can be removed.

File: obsei/analyzer/ner_analyzer.py
Comment (lalitpagaria) [inline obsei/analyzer/ner_analyzer.py:40]: Let's simplify whole code section as follows -
```
tokenizer = AutoTokenizer.from_pretrained(
          self.tokenizer_name if self.tokenizer_name else self.model_name_or_path, use_fast=True
)
```

File: obsei/analyzer/ner_analyzer.py
Comment (lalitpagaria) [inline obsei/analyzer/ner_analyzer.py:85]: Interesting. Thanks for pointing it out.

Ideally `batch_size` should be part of Analyzer. Hence in this case `BaseAnalyzer` is correct place for this.
Let's move`batch_size` from `analyzer_confg` to `BaseAnalyzer` and correct this anomaly.

**Review by lalitpagaria — CHANGES_REQUESTED** [review]
@akar5h Thank you for working on it.
I have added few comments. Can you please address them.

File: obsei/analyzer/ner_analyzer.py
Comment (akar5h) [reply obsei/analyzer/ner_analyzer.py:85]: Isnt `BaseAnalyzerConfig` is inherited by other AnalyzerConfig class meant for configurations like batch size or labels. 
By moving batch_size to `BaseAnalyzer` will be more unclean , 

I suggest that we can add a BaseAnalyzerConfig as a class variable with batch size 1 in NerAnalyzer , this will be a cleaner way. 

```
class NERAnalyzer(BaseAnalyzer):
    _pipeline: Pipeline = PrivateAttr()
    _max_length: int = PrivateAttr()
    TYPE: str = "NER"
    model_name_or_path: str
    tokenizer_name: Optional[str] = None
    grouped_entities: Optional[bool] = True

    ner_config : BaseAnalyzerConfig = BaseAnalyzerConfig(batch_size=1)   # This line

```

and use this ner config in place of `analyzer_config`  in `analyze_input`

File: obsei/analyzer/ner_analyzer.py
Comment (akar5h) [reply obsei/analyzer/ner_analyzer.py:40]: Yes, this is much cleaner

File: obsei/analyzer/ner_analyzer.py
Comment (lalitpagaria) [reply obsei/analyzer/ner_analyzer.py]: batch_size is more of infra property instead of configuration ie labels, hence better suited near model name and other parameters. For example if transformers model inherent provide batching then Obsei might skip batching at all. Also value of batch size will be depend on where we are running model via analyzer ie system of high or low memory?

Let's assume ClassificationAnalyzer running on CPU and GPU will use different batch_size but same analyzer_config.

File: obsei/analyzer/ner_analyzer.py
Comment (lalitpagaria) [inline obsei/analyzer/ner_analyzer.py:34]: please run [black formatting](https://github.com/lalitpagaria/obsei/blob/master/CONTRIBUTING.md#submitting-a-pull-request) on this code.

**Comment by akar5h** [comment]
@lalitpagaria I saw your comments and worked on them, batch size was moved to BaseAnalyzer, and ran the black refactoring. Any more suggestions ? Also Ill start looking into the workflow build error if it persists later tonight.


_Also, this is the first time I'm working on open source so will take a bit time to get the hang of the workflow and processes involved._

**Comment by lalitpagaria** [comment]
@akar5h Thank you very much for choosing Obsei for your first contribution. Don't worry about workflows, you can learn slowly, they make PR reviewer's and maintainer's life easy :)

BTW I have reviewed your PR again and it is in very good shape. I few these two comments -
1. Type checking is failing, refer this https://github.com/lalitpagaria/obsei/pull/152/checks?check_run_id=2945779483. I think you nee to keep this line as it is https://github.com/lalitpagaria/obsei/pull/152/files#diff-71ab2e5daf0fed8592831d6454f6e4087546f1684a41974aa09ece24f1cbfa06L78
2. Can you please remove `batch_size` from `BaseAnalyzerConfig`

Please let me know once you resolve these I will merge your PR.

**Comment by akar5h** [comment]
@lalitpagaria I have made these changes. Thanks for pointing in the right direction

**Review by lalitpagaria — APPROVED** [review]
@akar5h Changes looks good to me.
Thank you for working on it and choosing Obsei for your first open source contribution.

I will merge after CI jobs completion.

**Comment by lalitpagaria** [comment]
@akar5h it seems with batch_size removal from BaseAnalyzerConfig, all tests are failing. Is it possible for you to fix them?

**Comment by akar5h** [comment]
Hi , @lalitpagaria I can try and fix them . Is it also possible for me to run the CI tests ? It would easier for me to debug that way and it would not be such a back and forth process, I just test them on some simple examples on my local rn,

**Comment by akar5h** [comment]
Ill refer to the logs and fix them

**Comment by lalitpagaria** [comment]
@akar5h it seems contribution guideline is missing this information. I will add this when fine time. Meanwhile you can locally run tests as follows
```
pip install pytest
cd test
pytest
```

**Comment by akar5h** [comment]
@lalitpagaria thanks for the pytest tip, the whole testing part became more clear now, 

The same batch_size changes were required to be done in `translation_analyzer` . 
Also can you review changes in test scripts .  earlier batch_size was passed as a `BaseAnalyzerConfig` object in test files, that was also changed . And in the NER test files, 
One of the test contained name "Lalit" which the tokenizer split into  _Lal## , ##it_ , giving them both **"PER"** tag.  Now to make the test consistent across different tokenizer , I changed the test input.  Can you review that too .

With these changes all the tests are passing

**Review by lalitpagaria — APPROVED** [review]
Great! LGTM!

Thanks for finding the issue, fixing it and improving current tests

**Pull request merged**
